src/main/starlark/builtins_bzl/common/cc/cc_toolchain.bzl

```diff
@@ -342,5 +344,15 @@ The label of the rule providing <code>cc_toolchain_config_info</code>.""",
             default = semantics.BUILD_INFO_TRANLATOR_LABEL,
             providers = [OutputGroupInfo],
         ),
+        "_aggregate_ddi": attr.label(
```

Comment:
Instead of putting the attr.label here directly, in semantics.bzl, have, e.g. _get_aggregate_ddi() which returns this attr.label block.

Then for folks who use semantics who don't support this (yet), they don't need to provide the tool at all. (e.g. they can return an empty `attr.label`).

Reply:
@comius unless you want to just refactor this on our end when you do the import?

Reply:
@trybka
I updated the code
add `_get_aggregate_ddi` and `_get_generate_modmap` in `semantics.bzl`